test/e2e/scale.go

```diff
@@ -79,6 +83,10 @@ type scaleSpecInput struct {
 	// Can be overridden by variable CAPI_SCALE_CLUSTER_COUNT.
 	ClusterCount *int64
 
+	// SeparateNamespacePerCluster defines if each cluster should be deployed into its separate namespace.
+	// In this case The namespace name will be the name of the cluster.
+	SeparateNamespacePerCluster bool
```

Comment:
nit: what about
```suggestion
	CreateNamespaceForEachCluster bool
```

Reply:
I went with `DeployClusterInSeparateNamespaces`. I think the most important part is that we deploy the clusters in separate namespaces. CreateNamespaceForEachCluster sounds like we just create a namespace for each cluster.